metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -2314,11 +2315,24 @@ public CompletableFuture<UpdateFeaturesResponseData> updateFeatures(
         }).thenApply(result -> {
             UpdateFeaturesResponseData responseData = new UpdateFeaturesResponseData();
             responseData.setResults(new UpdateFeaturesResponseData.UpdatableFeatureResultCollection(result.size()));
-            result.forEach((featureName, error) -> responseData.results().add(
-                new UpdateFeaturesResponseData.UpdatableFeatureResult()
-                    .setFeature(featureName)
-                    .setErrorCode(error.error().code())
-                    .setErrorMessage(error.message())));
+            Optional<Entry<String, ApiError>> errorEntry = result.entrySet().stream().filter(entry ->
+                !entry.getValue().error().equals(Errors.NONE)).findFirst();
+                
+            if (errorEntry.isPresent()) {
+                String errorFeatureName = errorEntry.get().getKey();
+                ApiError topError = errorEntry.get().getValue();
+                String errorString = errorFeatureName + ":" + topError.error().exceptionName() + " (" + topError.message() + ")";
```

Comment:
Do we need to construct the new error message? `FeatureControlManager.invalidUpdateVersion` already includes the feature name in the error message.

Reply:
I think I originally did this to also include the exception name but if we just take it as the top level exception we don't need to do this.